pkg/windows/templates.go

```diff
@@ -0,0 +1,130 @@
+// +build windows
+
+package windows
+
+import (
+	"bytes"
+	"html/template"
+	"net/url"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+)
+
+const CalicoConfigTemplate = `{
+  "name": "{{ .CalicoConfig.Name }}",
+  "windows_use_single_network": true,
+  "cniVersion": "{{ .CalicoConfig.Cni.Version }}",
+  "type": "calico",
+  "mode": "{{ .CalicoConfig.Mode }}",
+  "vxlan_mac_prefix":  "{{ .CalicoConfig.Felix.MacPrefix }}",
+  "vxlan_vni": {{ .CalicoConfig.Felix.Vxlanvni }},
+  "policy": {
+    "type": "k8s"
+  },
+  "log_level": "info",
+  "capabilities": {"dns": true},
+  "DNS": {
+    "Nameservers": [
+		"{{ .CalicoConfig.DnsServers }}"
+    ],
+    "Search":  [
+      "svc.cluster.local"
+    ]
+  },
+  "nodename_file": "{{ replace .CalicoConfig.NodeNameFile }}",
+  "datastore_type": "{{ .CalicoConfig.DatastoreType }}",
+  "etcd_endpoints": "{{ .CalicoConfig.EtcdEndpoints }}",
+  "etcd_key_file": "{{ .CalicoConfig.EtcdKeyFile }}",
+  "etcd_cert_file": "{{ .CalicoConfig.EtcdCertFile }}",
+  "etcd_ca_cert_file": "{{ .CalicoConfig.EtcdCaCertFile }}",
+  "kubernetes": {
+    "kubeconfig": "{{ replace .CalicoConfig.KubeConfig.Path }}"
+  },
+  "ipam": {
+    "type": "{{ .CalicoConfig.Cni.IpamType }}",
+    "subnet": "usePodCidr"
+  },
+  "policies":  [
+    {
+      "Name":  "EndpointPolicy",
+      "Value":  {
+        "Type":  "OutBoundNAT",
+        "ExceptionList":  [
+          "{{ .CalicoConfig.ServiceCidr }}"
+        ]
+      }
+    },
+    {
+      "Name":  "EndpointPolicy",
+      "Value":  {
+        "Type":  "SDNROUTE",
+        "DestinationPrefix":  "{{ .CalicoConfig.ServiceCidr }}",
+        "NeedEncap":  true
+      }
+    }
+  ]
+}
+`
+
+const CalicoKubeConfigTemplate = `apiVersion: v1
+kind: Config
+clusters:
+- name: kubernetes
+  cluster:
+    certificate-authority: {{ .CalicoConfig.KubeConfig.CertificateAuthority }}
+    server: {{ .CalicoConfig.KubeConfig.Server }}
+contexts:
+- name: calico-windows@kubernetes
+  context:
+    cluster: kubernetes
+    namespace: kube-system
+    user: calico-windows
+current-context: calico-windows@kubernetes
+users:
+- name: calico-windows
+  user:
+    token: {{ .CalicoConfig.KubeConfig.Token }}
+`
+
+// ParseTemplateFromConfig takes a template and CniConfig and generates a a windows specific
+// CNI config.
+func ParseTemplateFromConfig(templateBuffer string, config *CniConfig) (string, error) {
+	out := new(bytes.Buffer)
```

Comment:
`new` is generally not used and the Go devs are considering removing it. We should avoid it's use for now.